services/app/src/contexts/ChessBoardContext.jsx

```diff
@@ -108,17 +110,22 @@ const ChessBoardContextProvider = ({ children }) => {
     } else {
       setGlobalScore(roundUp(avg))
     }
-    setBoard(copyOfBoard);
+    chessBoard.setBoard(copyOfBoard);
   };
 
   const clearBoardItems = () => {
     setScore(0);
     setGlobalScore(0);
-    setBoard(defaultBoard);
+    chessBoard.clearChessBoard();
     setLastTurn(null);
     setSelectedItem('');
   };
 
+  const removeFigureFromBoard = () => {
+    setLastTurn(null);
```

Comment:
const lastTurn = useMemo(() => {
    const allTurns = chessBoard.board
      .flatMap(row =>
        row.filter(cellLike => 'items' in cellLike)
          .flatMap(cell => cell.items)
      );

    return allTurns.find(turn => turn.playerId === currentPlayerId) ?? null;
  }, [chessBoard.board, currentPlayerId]);